test/E2ETest/DockerAssert.cs

```diff
@@ -76,6 +87,34 @@ void OnOutput(string text)
             }
         }
 
+        private static async Task AssertAboutContainersAsync(ITestOutputHelper output, string[] containers, Func<HashSet<string>, string[], bool> predicate, Func<string[], string> errorText)
```

Comment:
Do we need all of this complex layering with the error message and predicate?

It seems like this could just be a method returns `IEnumerable<string>`of the containers, and then the validation and error message building wouldn't have to be packed into lambdas.